core/src/test/scala/org/apache/spark/crypto/ShuffleEncryptionSuite.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.crypto
+
+import java.security.PrivilegedExceptionAction
+
+import org.apache.hadoop.security.{Credentials, UserGroupInformation}
+
+import org.apache.spark.{SparkConf, SparkFunSuite}
+import org.apache.spark.crypto.CryptoConf._
+import org.apache.spark.crypto.CryptoStreamUtils.SPARK_CHIMERA_CONF_PREFIX
+
+private[spark] class ShuffleEncryptionSuite extends SparkFunSuite {
+  val ugi = UserGroupInformation.createUserForTesting("testuser", Array("testgroup"))
+
+  test("Test Chimera configuration conversion") {
```

Comment:
I haven't checked what the Chimera conf looks like lately, but I think it goes without saying that this code should be creating a valid Chimera conf, regardless of how it achieves that.

It would be better to not have to change Chimera for something like this, but since it's a relatively new library with a somewhat in-development public interface, I think it's ok to change it now. But note that once there's code depending on the behavior, it cannot be changed anymore.